File: Examples/VisualBasic/CreateAndManageTwoDBarcodes/UseAztecErrorCorrection.vb
Imports Aspose.BarCode.Generation

'
' This project uses Automatic Package Restore feature of NuGet to resolve Aspose.BarCode for .NET API reference 
' When the project is build. Please check https://docs.nuget.org/consume/nuget-faq for more information. 
' If you do not wish to use NuGet, you can manually download Aspose.BarCode for .NET API from http://www.aspose.com/downloads, 
' Install it and then add its reference to this project. For any issues, questions or suggestions 
' Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.BarCode.Examples.VisualBasic.CreateAndManageTwoDBarcodes
    Class UseAztecErrorCorrection
        Public Shared Sub Run()
            'ExStart:UseAztecErrorCorrection
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_CreateAndManage2DBarCodes()

            ' Instantiate barcode object and set Barcode CodeText & Symbology
            Dim barCodeBuilder As New BarCodeBuilder("123456789", EncodeTypes.Aztec) With { _
                .AztectErrorLevel = 10 _
            }
            barCodeBuilder.Save(dataDir & Convert.ToString("UseAztecErrorCorrection_out.bmp"), BarCodeImageFormat.Bmp)
            'ExEnd:UseAztecErrorCorrection
        End Sub
    End Class
End Namespace

File: Examples/CSharp/CreateAndManageTwoDBarcodes/UseAztecErrorCorrection.cs
using System.IO;
using System;
using Aspose.BarCode;

/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.BarCode for .NET API reference 
when the project is build. Please check https:// ocs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.BarCode for .NET API from http:// ww.aspose.com/downloads, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using http:// ww.aspose.com/community/forums/default.aspx
*/

namespace Aspose.BarCode.Examples.CSharp.CreateAndManageTwoDBarcodes
{
    class UseAztecErrorCorrection
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_CreateAndManage2DBarCodes();

            // Instantiate barcode object
            BarCodeBuilder barCodeBuilder = new BarCodeBuilder();
            
            // Set Barcode Symbology
            barCodeBuilder.SymbologyType = Symbology.Aztec;

            // Set Error Error Correction level
            barCodeBuilder.AztectErrorLevel = 10;

            // Set CodeText 
            barCodeBuilder.CodeText = "123456789";

            // Save the Barcode image
            barCodeBuilder.Save(dataDir + "UseAztecErrorCorrection_out.bmp", BarCodeImageFormat.Bmp);

            Console.WriteLine(Environment.NewLine + "Barcode saved at " + dataDir + "UseAztecErrorCorrection_out.bmp");
        }
    }
}
